src/dune_engine/action_builder.ml

```diff
@@ -277,67 +277,6 @@ let progn ts =
   let+ actions = With_targets.all ts in
   Action.Progn actions
 
-(* Analysis *)
-
-module rec Analysis : sig
-  val static_deps : _ t -> Static_deps.t
-end = struct
-  module Input = struct
-    type t = T : _ memo -> t
-
-    let equal (T x) (T y) = Type_eq.Id.equal x.id y.id
-
-    let hash (T m) = Type_eq.Id.hash m.id
-
-    let to_dyn (T m) = Dyn.String m.name
-  end
-
-  let memo =
-    Memo.create_hidden "Action_builder.static_deps"
-      ~input:(module Input)
-      Sync
-      (fun (T m) -> Analysis.static_deps m.t)
-
-  let rec static_deps : type a. a t -> Static_deps.t =
-   fun t ->
-    let file_exists = Fdecl.get file_exists_fdecl in
```

Comment:
Do we still use `file_exists` and the corresponding `Fdecl` and constructor anywhere? Perhaps, this can also be simplified away.

Reply:
Oh, we do use the constructor (missed on the first pass), but do we still need the `Fdecl`? Perhaps, we can now pass it directly.

Reply:
Good point. I added it to the functor.

Reply:
:+1: